webapp/src/Controller/API/GeneralInfoController.php

```diff
@@ -256,6 +271,158 @@ public function getDatabaseConfigurationAction(Request $request)
         return $result;
     }
 
+    /**
+     * Update configuration variables
+     * @Rest\Put("/config")
+     * @SWG\Response(
+     *     response="200",
+     *     description="The full configuration after change",
+     *     @SWG\Schema(type="object")
+     * )
+     * @SWG\Parameter(
+     *     name="config",
+     *     in="formData",
+     *     type="object",
+     *     description="The config variables to update",
+     *     required=true
+     * )
+     * @param Request $request
+     * @return \App\Entity\Configuration[]|mixed
+     * @throws \Exception
+     */
+    public function updateConfigurationAction(Request $request)
+    {
```

Comment:
It feels like that a lot of this code actually belongs in the `ConfigurationService`. Also, there's a lot of duplication with `Jury/ConfigController.php`.

So I'd prefer seeing what common code can be moved into the `ConfigurationService`.

Reply:
It might indeed be better to do the deduplication before we introduce it instead of in the future

Reply:
I moved the shared code to one spot. It required to inject some services (DOMjudgeService, EventLogService) as arguments, since they could not be autowired in the constructor, since they both need the config service and that results in a circular dependency.

Also added admin security.